Consuming messages compressed with zstd doesn't work

When using the native executable and Direct connections to consume messages from a Kafka topic that holds records that were compressed with `zstd`, we run into the following error:

```
2025-01-17 21:20:58,423 c7mw2pkhk6 /Users/stefansprenger/.vscode/extensions/confluentinc.vscode-confluent-0.23.3-darwin-arm64/ide-sidecar-0.134.1-runner[20538] ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (executor-thread-857) HTTP Request to /gateway/v1/clusters/oh-sxaDRTcyAr6pFRbXyzA/topics/zstd_compression/partitions/-/consume failed, error id: 6f34130f-4d48-41d4-8dd9-3c1cd644fcc8-2: org.apache.kafka.common.KafkaException: Received exception when fetching the next record from zstd_compression-0. If needed, please seek past the record to continue consumption.
        at org.apache.kafka.clients.consumer.internals.CompletedFetch.fetchRecords(CompletedFetch.java:299)
        at org.apache.kafka.clients.consumer.internals.FetchCollector.fetchRecords(FetchCollector.java:168)
        at org.apache.kafka.clients.consumer.internals.FetchCollector.collectFetch(FetchCollector.java:134)
        at org.apache.kafka.clients.consumer.internals.Fetcher.collectFetch(Fetcher.java:145)
        at org.apache.kafka.clients.consumer.internals.LegacyKafkaConsumer.pollForFetches(LegacyKafkaConsumer.java:693)
        at org.apache.kafka.clients.consumer.internals.LegacyKafkaConsumer.poll(LegacyKafkaConsumer.java:617)
        at org.apache.kafka.clients.consumer.internals.LegacyKafkaConsumer.poll(LegacyKafkaConsumer.java:590)
        at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:874)
        at io.confluent.idesidecar.restapi.messageviewer.SimpleConsumer.consume(SimpleConsumer.java:91)
        at io.confluent.idesidecar.restapi.messageviewer.strategy.NativeConsumeStrategy.consumeMessages(NativeConsumeStrategy.java:59)
        at io.confluent.idesidecar.restapi.messageviewer.strategy.NativeConsumeStrategy.lambda$execute$0(NativeConsumeStrategy.java:35)
        at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$0(ContextImpl.java:178)
        at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:270)
        at io.vertx.core.impl.ContextImpl.lambda$internalExecuteBlocking$2(ContextImpl.java:210)
        at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:76)
        at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:635)
        at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base@21.0.2/java.lang.Thread.runWith(Thread.java:1596)
        at java.base@21.0.2/java.lang.Thread.run(Thread.java:1583)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:833)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.posix.thread.PosixPlatformThreads.pthreadStartRoutine(PosixPlatformThreads.java:211)
Caused by: org.apache.kafka.common.KafkaException: java.lang.UnsatisfiedLinkError: com.github.luben.zstd.ZstdInputStreamNoFinalizer.createDStream()J [symbol: Java_com_github_luben_zstd_ZstdInputStreamNoFinalizer_createDStream or Java_com_github_luben_zstd_ZstdInputStreamNoFinalizer_createDStream__]
        at org.apache.kafka.common.compress.ZstdFactory.wrapForInput(ZstdFactory.java:70)
        at org.apache.kafka.common.record.CompressionType$5.wrapForInput(CompressionType.java:155)
        at org.apache.kafka.common.record.DefaultRecordBatch.recordInputStream(DefaultRecordBatch.java:273)
        at org.apache.kafka.common.record.DefaultRecordBatch.compressedIterator(DefaultRecordBatch.java:277)
        at org.apache.kafka.common.record.DefaultRecordBatch.streamingIterator(DefaultRecordBatch.java:358)
        at org.apache.kafka.clients.consumer.internals.CompletedFetch.nextFetchedRecord(CompletedFetch.java:223)
        at org.apache.kafka.clients.consumer.internals.CompletedFetch.fetchRecords(CompletedFetch.java:274)
        ... 25 more
Caused by: java.lang.UnsatisfiedLinkError: com.github.luben.zstd.ZstdInputStreamNoFinalizer.createDStream()J [symbol: Java_com_github_luben_zstd_ZstdInputStreamNoFinalizer_createDStream or Java_com_github_luben_zstd_ZstdInputStreamNoFinalizer_createDStream__]
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.access.JNINativeLinkage.getOrFindEntryPoint(JNINativeLinkage.java:152)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.JNIGeneratedMethodSupport.nativeCallAddress(JNIGeneratedMethodSupport.java:54)
        at com.github.luben.zstd.ZstdInputStreamNoFinalizer.createDStream(Native Method)
        at com.github.luben.zstd.ZstdInputStreamNoFinalizer.<init>(ZstdInputStreamNoFinalizer.java:69)
        at org.apache.kafka.common.compress.ZstdFactory.wrapForInput(ZstdFactory.java:68)
        ... 31 more
```

--- a/src/main/resources/META-INF/native-image/com.github.luben/zstd-jni/index.json
+++ b/src/main/resources/META-INF/native-image/com.github.luben/zstd-jni/index.json
@@ -1,0 +1,5 @@
+[
+  "jni-config.json",
+  "reflect-config.json",
+  "resource-config.json"
+]

--- a/src/main/resources/META-INF/native-image/com.github.luben/zstd-jni/jni-config.json
+++ b/src/main/resources/META-INF/native-image/com.github.luben/zstd-jni/jni-config.json
@@ -1,0 +1,100 @@
+[
+  {
+    "condition": {
+      "typeReachable": "com.github.luben.zstd.ZstdCompressCtx"
+    },
+    "name": "com.github.luben.zstd.ZstdCompressCtx",
+    "fields": [
+      {
+        "name": "nativePtr"
+      }
+    ]
+  },
+  {
+    "condition": {
+      "typeReachable": "com.github.luben.zstd.ZstdDecompressCtx"
+    },
+    "name": "com.github.luben.zstd.ZstdDecompressCtx",
+    "fields": [
+      {
+        "name": "nativePtr"
+      }
+    ]
+  },
+  {
+    "condition": {
+      "typeReachable": "com.github.luben.zstd.ZstdDictCompress"
+    },
+    "name": "com.github.luben.zstd.ZstdDictCompress",
+    "fields": [
+      {
+        "name": "nativePtr"
+      }
+    ]
+  },
+  {
+    "condition": {
+      "typeReachable": "com.github.luben.zstd.ZstdDictDecompress"
+    },
+    "name": "com.github.luben.zstd.ZstdDictDecompress",
+    "fields": [
+      {
+        "name": "nativePtr"
+      }
+    ]
+  },
+  {
+    "condition": {
+      "typeReachable": "com.github.luben.zstd.ZstdDirectBufferCompressingStreamNoFinalizer"
+    },
+    "name": "com.github.luben.zstd.ZstdDirectBufferCompressingStream$1"
+  },
+  {
+    "condition": {
+      "typeReachable": "com.github.luben.zstd.ZstdDirectBufferCompressingStreamNoFinalizer"
+    },
+    "name": "com.github.luben.zstd.ZstdDirectBufferCompressingStreamNoFinalizer",
+    "fields": [
+      {
+        "name": "consumed"
+      },
+      {
+        "name": "produced"
+      }
+    ]
+  },
+  {
+    "condition": {
+      "typeReachable": "com.github.luben.zstd.ZstdDirectBufferDecompressingStreamNoFinalizer"
+    },
+    "name": "com.github.luben.zstd.ZstdDirectBufferDecompressingStream$1"
+  },
+  {
+    "condition": {
+      "typeReachable": "com.github.luben.zstd.ZstdDirectBufferDecompressingStreamNoFinalizer"
+    },
+    "name": "com.github.luben.zstd.ZstdDirectBufferDecompressingStreamNoFinalizer",
+    "fields": [
+      {
+        "name": "consumed"
+      },
+      {
+        "name": "produced"
+      }
+    ]
+  },
+  {
+    "condition": {
+      "typeReachable": "com.github.luben.zstd.ZstdInputStreamNoFinalizer"
+    },
+    "name": "com.github.luben.zstd.ZstdInputStreamNoFinalizer",
+    "fields": [
+      {
+        "name": "dstPos"
+      },
+      {
+        "name": "srcPos"
+      }
+    ]
+  }
+]

--- a/src/main/resources/META-INF/native-image/com.github.luben/zstd-jni/reflect-config.json
+++ b/src/main/resources/META-INF/native-image/com.github.luben/zstd-jni/reflect-config.json
@@ -1,0 +1,38 @@
+[
+  {
+    "condition": {
+      "typeReachable": "com.github.luben.zstd.util.Native"
+    },
+    "name": "java.security.SecureRandomParameters"
+  },
+  {
+    "condition": {
+      "typeReachable": "com.github.luben.zstd.util.Native"
+    },
+    "name": "org.osgi.framework.BundleEvent"
+  },
+  {
+    "condition": {
+      "typeReachable": "com.github.luben.zstd.util.Native"
+    },
+    "name": "sun.security.provider.NativePRNG",
+    "methods": [
+      {
+        "name": "<init>",
+        "parameterTypes": []
+      }
+    ]
+  },
+  {
+    "condition": {
+      "typeReachable": "com.github.luben.zstd.util.Native"
+    },
+    "name": "sun.security.provider.SHA",
+    "methods": [
+      {
+        "name": "<init>",
+        "parameterTypes": []
+      }
+    ]
+  }
+]

--- a/src/main/resources/META-INF/native-image/com.github.luben/zstd-jni/resource-config.json
+++ b/src/main/resources/META-INF/native-image/com.github.luben/zstd-jni/resource-config.json
@@ -1,0 +1,13 @@
+{
+  "bundles": [],
+  "resources": {
+    "includes": [
+      {
+        "condition": {
+          "typeReachable": "com.github.luben.zstd.util.Native"
+        },
+        "pattern": "[a-z]+/[a-z0-9]+/libzstd-jni-[0-9.-]+(so|dll|dylib)"
+      }
+    ]
+  }
+}

--- a/src/main/resources/application.yml
+++ b/src/main/resources/application.yml
@@ -236,7 +236,7 @@ quarkus:
   jackson:
     fail-on-unknown-properties: true
   native:
-    additional-build-args: --initialize-at-run-time=io.confluent.idesidecar.restapi.auth.CCloudOAuthConfig\,io.confluent.idesidecar.restapi.auth.CCloudOAuthContext\,io.confluent.ide.sidecar.restapi.util.WebClientFactory\,org.apache.kafka.common.security.authenticator.SaslClientAuthenticator\,org.apache.avro.file.DataFileWriter\,io.confluent.kafka.schemaregistry.avro.AvroSchemaUtils
+    additional-build-args: --initialize-at-run-time=io.confluent.idesidecar.restapi.auth.CCloudOAuthConfig\,io.confluent.idesidecar.restapi.auth.CCloudOAuthContext\,io.confluent.ide.sidecar.restapi.util.WebClientFactory\,org.apache.kafka.common.security.authenticator.SaslClientAuthenticator\,org.apache.avro.file.DataFileWriter\,io.confluent.kafka.schemaregistry.avro.AvroSchemaUtils\,com.github.luben.zstd
     resources:
       includes: templates/callback.html,templates/callback_failure.html,static/templates.zip,feature-flag-defaults.json
   smallrye-health:

